scripts/benchmark/utils/reporting.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from typing import (
+    NamedTuple
+)
+
+from utils.shellart import (
+    bold_white
+)
+
+
+class DefaultStat(NamedTuple):
+    counter: int = 0
+    caption: str = ''
+    total_blocks: int = 0
+    total_seconds: int = 0
+    total_gas: int = 0
+
+    @property
+    def blocks_per_second(self) -> float:
+        return self.total_blocks / self.total_seconds
+
+    @property
+    def gas_per_second(self) -> float:
+        return self.total_gas / self.total_seconds
+
+    @property
+    def avg_total_blocks(self) -> float:
+        return self.total_blocks / self.counter
+
+    @property
+    def avg_total_seconds(self) -> float:
+        return self.total_seconds / self.counter
+
+    @property
+    def avg_total_gas(self) -> float:
+        return self.total_gas / self.counter
+
+    def cumulate(self, stat: 'DefaultStat') -> 'DefaultStat':
+        return DefaultStat(
+            counter=self.counter + 1,
+            total_blocks=self.total_blocks + stat.total_blocks,
+            total_seconds=self.total_seconds + stat.total_seconds,
+            total_gas=self.total_gas + stat.total_gas,
+        )
+
+
+REPORT_TABLE_LENGTH = 110
+SINGLE_UNDERLINE = '-' * REPORT_TABLE_LENGTH
+DOUBLE_UNDERLINE = '=' * REPORT_TABLE_LENGTH
+HASH_UNDERLINE = '#' * REPORT_TABLE_LENGTH
+
+
+def print_default_benchmark_result_header() -> None:
+    logging.info(SINGLE_UNDERLINE)
+    logging.info('|{:^19}|{:^16}|{:^16}|{:^20}|{:^16}|{:^16}|'.format(
+        'VM',
+        'total blocks',
+        'total seconds',
+        'blocks / second',
+        'total gas',
+        'gas / second'
+    ))
+    logging.info(SINGLE_UNDERLINE)
+
+
+def print_default_benchmark_stat_line(stat: DefaultStat) -> None:
+    logging.info(
+        '|{stat.caption:^19}'
+        '|{stat.total_blocks:^16}'
+        '|{stat.total_seconds:^16.3f}'
+        '|{stat.blocks_per_second:^20.3f}'
+        '|{stat.total_gas:^16,}'
+        '|{stat.gas_per_second:^16,.3f}'
+        '|'.format(
+            stat=stat
+        ))
+
+
+def print_default_benchmark_total_line(stat: DefaultStat) -> None:
+    logging.info(SINGLE_UNDERLINE)
+    logging.info(bold_white(
+        '|{caption:^19}'
```

Comment:
Not a big deal at all, and not something we should try to do in this PR, but we could potentially:

1. Use a special class of [`LogRecord`](https://docs.python.org/3/library/logging.html#logrecord-objects)
2. Use the contextual data APIs: https://docs.python.org/3/howto/logging-cookbook.html#context-info